### PR TITLE
plan: remove IsCorrelated and SetCorrelated.

### PR DESCRIPTION
**Pull request by hanfei1991**
clean code
@shenli @coocood @winoros @zimulala PTAL

### CODE REVIEW

File: plan/optimizer.go [hunk 1]
```diff
@@ -82,7 +82,6 @@ func doOptimize(logic LogicalPlan, ctx context.Context, allocator *idAllocator)
 	}
 	pp := info.p
 	pp = EliminateProjection(pp)
-	physicalInitialize(pp)
 	addCachePlan(pp, allocator)
```
Comment (zimulala) [inline plan/optimizer.go]: Will it be too late to deal with correlated columns here？

Comment (hanfei1991) [inline plan/optimizer.go]: 1. `Cache` Plan is a PhysicalPlan, we can only do it after physicalplanbuilder
2. The order of `EliminateProj` and `addCachePlan` is arbitrary.

Comment (zimulala) [inline plan/optimizer.go]: I mean to deal with correlated columns rather than deal with `addCachePlan`.

**Comment by zimulala** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**